hdrass/CCW+Rotator_integration_testing/Rotator_interlock_test.ipynb

```diff
@@ -0,0 +1,1007 @@
+    "\n",
+    "await rot.cmd_move.set_start(position=0.0,timeout=30.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await salobj.set_summary_state(mtm, salobj.State.STANDBY)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Wait for override off"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await salobj.set_summary_state(mtm, salobj.State.ENABLED)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Set Low Velocity and Move Through Negative Interlock"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "#The velocitySet command gets accepted but the moveConstantVelocity command gets error.\n",
+    "#moveConstantVelocity has not been implemented\n",
+    "\n",
+    "await rot.cmd_velocitySet.set_start(velocity=-0.02,moveDuration=115.)\n",
+    "\n",
+    "await asyncio.sleep(0.1)\n",
+    "\n",
+    "await rot.cmd_moveConstantVelocity.start(timeout=30.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "CCWstate = await mtm.evt_cameraCableWrapFollowing.aget(timeout=10.)\n",
+    "print(\"CCWstate:\",CCWstate.enabled)\n",
+    "await mtm.cmd_disableCameraCableWrapFollowing.set_start(timeout=10.)\n",
+    "CCWstate = await mtm.evt_cameraCableWrapFollowing.aget(timeout=10.)\n",
+    "print(\"CCWstate:\",CCWstate.enabled)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await asyncio.sleep(1.)\n",
+    "\n",
+    "print(\"Move to -10 deg position\")\n",
+    "\n",
+    "await rot.cmd_move.set_start(position=-10,timeout=30.)\n",
+    "\n",
+    "print(\"Reset the Rotator and CCW\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Bring the Rotator and CCW back to enabled state"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await mtm.cmd_clearerror.start(timeout=30)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Wait for override off"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await rot.cmd_clearError.start(timeout=30)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "CCWstate = await mtm.evt_cameraCableWrapFollowing.aget(timeout=10.)\n",
+    "print(\"CCWstate:\",CCWstate.enabled)\n",
+    "await mtm.cmd_enableCameraCableWrapFollowing.set_start(timeout=10.)\n",
+    "CCWstate = await mtm.evt_cameraCableWrapFollowing.aget(timeout=10.)\n",
```

Comment:
To my eyes this code seems to suggest that you want to wait for the new value after calling mtm.cmd_enableCameraCableWrapFollowing.set_start. But the code doesn't do that!

Repeated calls to `aget` are grossly misleading and *very strongly not recommended*. The first one may wait for an event. All subsequent calls are just a synonym for `get`, except you have to await the result.

If you want the current value then call `get` (without the await). If you want a new value then call `next` and wait for it.

I recommend *never* using `aget` because it is too misleading.

Reply:
If you call `aget` the first time then you can't be sure the buffer is empty so, in that situation I recommend:

call topic.aget or get to get the initial value
call topic.flush to flush the queue
do the operation that produces a new event
call topic.next to get the new event

Reply:
All the `aget` are out. First time `get` is used to get the actual value and then `next` with `await` to get the value after a change was done.